libcontainer/configs/validate/validator.go

```diff
@@ -259,6 +260,39 @@ func cgroupsCheck(config *configs.Config) error {
 	return nil
 }
 
+func mountsStrict(config *configs.Config) error {
+	for _, m := range config.Mounts {
+		if !m.IsIDMapped() {
+			continue
+		}
+
+		if !m.IsBind() {
+			return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is supported only for mounts with the option 'bind'", m)
+		}
+		if config.RootlessEUID {
+			return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is not supported when runc is being launched with EUID != 0, needs CAP_SYS_ADMIN on the runc parent's user namespace", m)
+		}
+		if config.UidMappings == nil || config.GidMappings == nil {
+			return fmt.Errorf("invalid mount %+v: not yet supported to use GidMappings/UidMappings in a mount without also using a user namespace", m)
+		}
+		if !sameMapping(config.UidMappings, m.UIDMappings) {
+			return fmt.Errorf("invalid mount %+v: not yet supported for the mount uidMappings to be different than user namespace uidMapping", m)
+		}
+		if !sameMapping(config.GidMappings, m.GIDMappings) {
+			return fmt.Errorf("invalid mount %+v: not yet supported for the mount gidMappings to be different than user namespace gidMapping", m)
+		}
+		// For idmap mounts we require src and dst to be an absolute path.
+		if !filepath.IsAbs(m.Destination) {
+			return fmt.Errorf("invalid mount %+v: mount destination not absolute", m)
+		}
```

Comment:
We have the same check in mounts() check, which was downgraded to a warning by commit b31a9340f92bf02d7fc1a8a9513885561e2fbc47. As this was done 2 years ago, maybe it's time to act on this TODO item and upgrade it to a warning, in the anticipation of 1.2.0?

Reply:
@kolyshkin s/upgrade to a warning/upgrade to an error, I guess?

Reply:
Upgraded to an error